agent/command/s3_put_test.go

```diff
@@ -430,6 +430,64 @@ func TestS3LocalFilesIncludeFilterPrefix(t *testing.T) {
 	}
 }
 
+func TestFileUploadNaming(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	var err error
+
+	dir := t.TempDir()
+	require.NoError(t, os.Mkdir(filepath.Join(dir, "subDir"), 0755))
+	f, err := os.Create(filepath.Join(dir, "subDir", "bar"))
+	require.NoError(t, err)
+	require.NoError(t, f.Close())
+
+	s := s3put{
+		AwsKey:                        "key",
+		AwsSecret:                     "secret",
+		Bucket:                        "bucket",
+		BuildVariants:                 []string{},
+		ContentType:                   "content-type",
+		LocalFilesIncludeFilter:       []string{"*"},
+		Permissions:                   s3.BucketCannedACLPublicRead,
+		LocalFilesIncludeFilterPrefix: "",
+		RemoteFile:                    "remote",
+	}
+	require.NoError(t, os.Mkdir(filepath.Join(dir, "destination"), 0755))
+	opts := pail.LocalOptions{
+		Path: filepath.Join(dir, "destination"),
+	}
+	s.bucket, err = pail.NewLocalBucket(opts)
+	require.NoError(t, err)
+	comm := client.NewMock("http://localhost.com")
+	conf := &internal.TaskConfig{
+		Expansions:   &util.Expansions{},
+		Task:         &task.Task{Id: "mock_id", Secret: "mock_secret"},
+		Project:      &model.Project{},
+		WorkDir:      dir,
+		BuildVariant: &model.BuildVariant{},
+	}
+	logger, err := comm.GetLoggerProducer(ctx, client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret}, nil)
+	require.NoError(t, err)
+
+	require.NoError(t, s.Execute(ctx, comm, logger, conf))
+	attachedFiles := comm.AttachedFiles
+	expected := map[string]bool{
+		"remotebar":       false,
+		"remoteremotebar": false,
+	}
+
+	for _, file := range attachedFiles {
+		for _, file := range file {
```

Comment:
Can we call the variable something other than `file` since the outer loop already defines `file`?